File: src/CorporateMessengerWithAgile/Application/Dto/CompanyDto.cs
namespace Application.Dto
{
    public class CompanyDto : BaseDto
    {
        public string Title { get; set; } = null!;
        public IReadOnlyList<Guid> EmployeeIds { get; set; } = null!;
        public IReadOnlyList<Guid> PositionIds { get; set; } = null!;
        public IReadOnlyList<Guid> ProjectIds { get; set; } = null!;
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/CompanyDto.ts
// src/models/entity/CompanyDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { GuidSchema } from '../Guid';

export const CompanyDtoSchema = BaseDtoSchema.extend({
    title: z.string().min(1),
    employeeIds: z.array(GuidSchema).optional(),
    positionIds: z.array(GuidSchema).optional(),
    projectIds: z.array(GuidSchema).optional(),
});

export type CompanyDto = z.infer<typeof CompanyDtoSchema>;
